modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -264,7 +270,7 @@ private void deletePartition(AmazonS3Reference clientReference, List<String> par
         }
     }
 
-    private static DeleteObjectsRequest bulkDelete(S3BlobStore blobStore, List<String> blobs) {
+    private static DeleteObjectsRequest bulkDelete(S3BlobStore blobStore, List<String> blobs, OperationPurpose purpose) {
```

Comment:
nit: all other places have `purpose` as first arg, I'd like to keep it consistent, so can we move it to be first here too?